src/bridge/MemberListSyncer.ts

```diff
@@ -278,19 +277,28 @@ export class MemberListSyncer {
         // map the filtered rooms to a list of users to join
         // [Room:{reals:[uid,uid]}, ...] => [{uid,roomid}, ...]
         const entries: { roomId: string; displayName: string; userId: string; frontier: boolean}[] = [];
-        filteredRooms.forEach((roomInfo) => {
-            roomInfo.realJoinedUsers.forEach((uid, index) => {
+        const idleRegex = this.server.ignoreIdleUsersOnStartupExcludeRegex;
+        for (const roomInfo of filteredRooms) {
+            for (const uid of roomInfo.realJoinedUsers) {
+                if (this.server.ignoreIdleUsersOnStartup) {
+                    const idle = await this.ircBridge.activityTracker?.isUserOnline(
```

Comment:
What kind of effect will this have on the freenode bridge startup time?

Reply:
I'd imagine a ms or two per user, this function will short circuit if we have a last active time for users already set, which we will have for 99.9% of users. For users we have never seen before, it will do a lookup for whois/presence, or finally give up and say they are offline. For matrix.org, neither of those lookup options work so it will short circuit those and immediately return false.

Basically, not a lot of time spent per user.